tests/core/contracts/test_contract_buildTransaction.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+
+import pytest
+
+# Ignore warning in pyethereum 1.6 - will go away with the upgrade
+pytestmark = pytest.mark.filterwarnings("ignore:implicit cast from 'char *'")
+
+
+@pytest.fixture()
+def math_contract(web3, MathContract):
+    deploy_txn = MathContract.deploy()
+    deploy_receipt = web3.eth.getTransactionReceipt(deploy_txn)
+    assert deploy_receipt is not None
+    _math_contract = MathContract(address=deploy_receipt['contractAddress'])
+    return _math_contract
+
+
+def test_build_transaction_with_contract_no_arguments(web3, math_contract):
+    txn = math_contract.buildTransaction().increment()
+    assert txn == {
+        'to': math_contract.address,
+        'data': '0xd09de08a',
+        'value': 0,
+        'gas': 43120,
+        'gasPrice': 1,
+        'chainId': 1
+    }
+
+
+def test_build_transaction_with_contract_class_method(web3, MathContract, math_contract):
+    txn = MathContract.buildTransaction({'to': math_contract.address}).increment()
+    assert txn == {
+        'to': math_contract.address,
+        'data': '0xd09de08a',
+        'value': 0,
+        'gas': 43120,
+        'gasPrice': 1,
+        'chainId': 1
+    }
+
+
+def test_build_transaction_with_contract_default_account_is_set(web3, math_contract):
+    txn = math_contract.buildTransaction().increment()
+    assert txn == {
+        'to': math_contract.address,
+        'data': '0xd09de08a',
+        'value': 0,
+        'gas': 43120,
+        'gasPrice': 1,
+        'chainId': 1
+    }
+
+
+def test_build_transaction_with_contract_data_supplied_errors(web3, math_contract):
+    with pytest.raises(ValueError):
+        math_contract.buildTransaction({
+            'data': '0x000'
+        }).increment()
```

Comment:
TODO: I think we should remove this call to `increment()` since I *think* the error occurs within `buildTransaction`?

Reply:
Yes it does :slightly_smiling_face: good catch. I can submit a follow up PR? Let me know if there's anything else to add

Reply:
A new PR is the way to go :+1: